### ISSUE TITLE
Notebook not displaying any output

### ISSUE THREAD

**Issue by lukeyeager** [facebook/iTorch#14]
I just tried installing and running iTorch according to the Readme, and my iTorch notebook won't give me any output. 

It seems like the same problem [this guy](http://stackoverflow.com/questions/13036197/ipython-notebook-getting-output) was having with iPython, but the `--ip=localhost` trick doesn't fix it for me.

If I run `ipython notebook` and enter `1+2`, then I get an `Out [1]: 3`. But when I do the same in `itorch notebook`, I get no `Out [1]:` at all. 

Am I the only one with this problem? I'm sure my problem is something simple. Is there a user group where questions like this should be asked instead of through GitHub? Thanks!

**Comment by soumith** [comment]
did you have torch installed already on your machine? is it a newly torch, or has it been around for a while.

Also, did you use sudo to install itorch?

**Comment by lukeyeager** [comment]
Just installed torch today according to [this](http://torch.ch/docs/getting-started.html). Did not install with sudo (It's at `/home/username/torch/install/bin/th`).
* Ubuntu 14.04
* Python 2.7.6
* iPython 2.3.1
* LuaJIT 2.1.0-alpha

**Comment by soumith** [comment]
Thanks for the info Luke. That seems to a standard and well-tested
configuration. Can you give me the output in the terminal when you type:
itorch

**Comment by lukeyeager** [comment]
```
Torch 7.0  Copyright (C) 2001-2011 Idiap, NEC Labs, NYU
Torch 7.0  Copyright (C) 2001-2011 Idiap, NEC Labs, NYU
	
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
Type "copyright", "credits" or "license" for more information.

IPython 2.3.1 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

IPython profile: torch
```

**Comment by alisabri** [comment]
Hi,
I had the same problem as Luke. Traced the possible causes and then found the following part in "IOHandler.lua" a bit dodgy :

49   while rawpub_port == 0 do
50       local portnum_f = torch.DiskFile(arg[3],'r')
51       portnum_f:quiet()
52       rawpub_port = portnum_f:readInt()
53       portnum_f:close()
54   end

it does not check read errors (because of quite) so might get the wrong port number. After adding line 53 below, now it works. 

49   while rawpub_port == 0 do
50       local portnum_f = torch.DiskFile(arg[3],'r')
51       portnum_f:quiet()
52       rawpub_port = portnum_f:readInt()
53       if portnum_f:hasError() then  rawpub_port = 0 end
54       portnum_f:close()
55   end

Would it be the reason ?
Regards,

**Comment by lukeyeager** [comment]
Uh... I don't think I had a problem after all. I just didn't know how to use iPython. Apparently Shift+Enter is the way to go. Whoops, thanks for the help!

@alisabri, I don't know what your problem is, but I'm pretty sure it's not related to my n00bery, so I'm going to mark this as closed.

**Comment by mobiledevver** [comment]
I know this was closed but I am experiencing the exact same problem.
Like the OP, ipython notebook works fine for me.  
Howevever, with itorch notebook, no output gets displayed.  I also tried the --ip=localhost flag to no avail.

A couple of observations:
-When using ipython, the browser creates an initial  code cell when it launches.  This doesn't happen with iTorch.
-No notebooks show as running with iTorch.

Appreciate any thoughts on the topic.

**Comment by dbieber** [comment]
I briefly ran into the same issue. The problem for me turned out to be I had defined print to be something other than the print function (eg print="hello"). Then when I ran a command, itorch was trying to print the results using the print variable and it was failing. Restarting the kernel did the trick, and now I can see my output again. However, if I run a command starting with = (eg =1+2) then I don't see any output.
